File: Compose-Bookshelf-App/app/src/main/java/com/example/bookshelfapp/model/GoogleBookApiModel.kt
package com.example.bookshelfapp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class GoogleBookApiModel(
    val kind: String,
    val items: List<GoogleBookItemsModel>,
    val totalItems: Int
)

@Serializable
data class GoogleBookItemsModel(
    val kind: String,
    val id: String,
    @SerialName("etag")
    val eTag: String,
    val selfLink: String,
    val volumeInfo: VolumeInfoModel
)

@Serializable
data class VolumeInfoModel(
    val title: String,
    val authors: List<String>
)

File: SwiftUI-Bookshelf-App/Bookshelf app/Model/GoogleBookApiModel.swift
//
//  GoogleBookApiModel.swift
//  Bookshelf app
//
//  Created by Jaehwa Noh on 1/27/24.
//

struct GoogleBookApiModel: Decodable {
    let kind: String
    let items: [GoogleBookItemsModel]
    let totalItems: Int
}

struct GoogleBookItemsModel: Decodable {
    let kind: String
    let id: String
    let eTag: String
    let selfLink: String
    let volumeInfo: VolumeInfoModel
    
    enum CodingKeys: String, CodingKey {
        case kind
        case id
        case eTag = "etag"
        case selfLink
        case volumeInfo
    }
}

struct VolumeInfoModel: Decodable {
    let title: String
    let authors: [String]?
}
